### ISSUE TITLE
Continue Training on Fully Trained Model

### ISSUE THREAD

**Issue by code8865** [MIC-DKFZ/nnUNet#1745]
Hello, 

first of all, thank you for your amazing work.  nnUNet has been very helpful throughout my segmentation projects. 

I have a question regarding fully-trained models/tasks based on its chosen trainer. Is there a way in v2 to increase the epoch number of the tasks retrospectively to continue the training? I have tried to increase the epoch number of the trainer manually while choosing to continue training, but this would cause errors.

Thanks a lot and best regards.

**Comment by LIKP0** [comment]
Hi,
I think this can help you. Use `-c` to continue training.
![image](https://github.com/MIC-DKFZ/nnUNet/assets/56829276/d63794bb-90ed-4743-a13f-f373daf53d34)

**Comment by ancestor-mithril** [comment]
The `--c` option works only if the model is not fully trained (`num_epochs` was not reached yet). 
If the model was already fully trained, it is not recommended to increase the number of epochs in the trainer manually to continue training. This is because the Polynomial Learning Rate Scheduler  depends on `num_epochs` and changing it during training will affect the learning rate schedule and implicitly, the training.

The recommended solution is to use another trainer and restart the training from scratch. 
You could also try the "-pretrained_weights" parameter to pass the weights of the already trained model to perform a new training.

**Comment by code8865** [comment]
Thank you for your answers!

I am trying to avoid training my model from scratch as it has already been trained for 8k epochs. 

If I were to train my model from scratch, but using TL ("-pretrained_weights"), wouldn't my weights get messed up due to the initial high learning rate?

**Comment by TaWald** [comment]
Thanks @ancestor-mithril @LIKP0  for the help :)

I would expect your very fine-tuned init to suffer a lot when you have the high learning rate and start from scratch again, so I would agree with the previous suggestion of building your own trainer. You can try to set `self.initial_lr = your_learning_rate` very low and potentially look into learning rate warm up, to gently pick up training again.

More generally, in our experience training super long (except transformers) does not contribute to a really significant increase in model performance. While the `progress.png` loss curve might suggest so, increasing epochs usually does not squeeze out a lot more performance. Maybe it's the nature of the PolyLR schedule, but just increasing epochs does not do the trick for us.
